agents-common/src/main/java/org/apache/ranger/plugin/service/RangerBasePlugin.java

```diff
@@ -1084,7 +1084,7 @@ private RangerAdminClient getAdminClient() throws Exception {
 			try {
 				@SuppressWarnings("unchecked")
 				Class<RangerChainedPlugin> pluginClass   = (Class<RangerChainedPlugin>) Class.forName(className);
-				RangerChainedPlugin        chainedPlugin = pluginClass.getConstructor(RangerBasePlugin.class, String.class).newInstance(this, chainedService);
+				RangerChainedPlugin        chainedPlugin = pluginClass.getConstructor(RangerBasePlugin.class, String.class, String.class).newInstance(this, pluginConfig.getServiceType(), chainedService);
```

Comment:
serviceType should that of chainedService; pluginConfig.getServiceType() would be the serviceType of current plugin.

Reply:
@CanGuan  - the expectation is that the class that implements RangerChainedPlugin has a constructor that takes 2 arguments (RangerBasePlugin parent, String serviceName), and that constructor calls RangerBasePlugin constructor with 3 arguments (RangerBasePlugin parent, String serviceType, String serviceName).

Reply:
@mneethiraj Please have a look. The service type for chained plugin is loaded from parent plugin config.